src/t_stream.c

```diff
@@ -3119,26 +3172,24 @@ void xclaimCommand(client *c) {
     mstime_t deliverytime = -1;  /* -1 means IDLE/TIME options not given. */
     int force = 0;
     int justid = 0;
+    int mkgroup = 0;
 
-    if (o) {
-        if (checkType(c,o,OBJ_STREAM)) return; /* Type error. */
-        group = streamLookupCG(o->ptr,c->argv[2]->ptr);
+    if(c->argc > 6 && !strcasecmp(c->argv[c->argc-1]->ptr,"MKGROUP")) {
+        mkgroup = 1;
     }
-
-    /* No key or group? Send an error given that the group creation
-     * is mandatory. */
-    if (o == NULL || group == NULL) {
-        addReplyErrorFormat(c,"-NOGROUP No such key '%s' or "
-                              "consumer group '%s'", (char*)c->argv[1]->ptr,
-                              (char*)c->argv[2]->ptr);
-        return;
-    }
-
+    
     if (getLongLongFromObjectOrReply(c,c->argv[4],&minidle,
         "Invalid min-idle-time argument for XCLAIM")
         != C_OK) return;
     if (minidle < 0) minidle = 0;
 
+    if (o == NULL) {
+        addReplyErrorFormat(c, "-NOGROUP No such key '%s'",
+                                     (char*)c->argv[1]->ptr);
+        return;
+    } 
+    if (checkType(c,o,OBJ_STREAM)) return; /* Type error. */
```

Comment:
let's move all of that (and the lookupKey at the top of this function) to after the argument parsing.
no sense in looking up the key if we have a syntax error.